src/google/protobuf/stubs/stringpiece.h

```diff
@@ -214,6 +214,13 @@ class PROTOBUF_EXPORT StringPiece {
       : ptr_(str.data()), length_(0) {
     length_ = CheckSize(str.size());
   }
+ 
+  StringPiece(  // NOLINT(runtime/explicit)
```

Comment:
I'd have to think some more about whether this change will be OK. Eventually we want to switch to `absl::string_view` but adding a dependency on absl is not entirely trivial and we haven't had time to work on it much yet. I think doing this as a stopgap measure is probably OK, but it needs to be guarded with an ifdef so that it only appears for C++17 and higher (we still support C++11 so we can't unconditionally rely on std::string_view yet).

Reply:
I agree.  We do something like this already, but we use a guard that checks the C++ feature macro for string_view.  (See https://github.com/protocolbuffers/protobuf/blob/6051341ab21b4e1f75e91710ab1a8cc8fadda521/src/google/protobuf/map.h )

That is, this should be guarded with
`#if defined(__cpp_lib_string_view)`

And at the top of the file, the string_view header should be included:

```
#if defined(__cpp_lib_string_view)
#include <string_view>
#endif
```

Reply:
Sorry, I am using string_view for so long by now.
I though it was introduced in C++11...

Should be fixed now.

Reply:
NB: as I see, abseil-cpp [falls back](https://github.com/abseil/abseil-cpp/blob/f72972654b69085b1d58fde04c618b77d3ac6935/absl/strings/string_view.h#L46) to std::string_view if it is available.

I am not sure about the logic behind `ABSL_USES_STD_STRING_VIEW` define, not I am sure on how abseil is going to add starts_with / ends_with from C++20 in the fallback case.